x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/inference_base.ts

```diff
@@ -60,19 +61,53 @@ export abstract class InferenceBase<TInferResponse> {
   protected abstract readonly inferenceTypeLabel: string;
   protected inputField: string;
   protected readonly modelInputField: string;
-  private inputText$ = new BehaviorSubject<string[]>([]);
+  protected inputText$ = new BehaviorSubject<string[]>([]);
   private inferenceResult$ = new BehaviorSubject<TInferResponse[] | null>(null);
   private inferenceError$ = new BehaviorSubject<MLHttpFetchError | null>(null);
   private runningState$ = new BehaviorSubject<RUNNING_STATE>(RUNNING_STATE.STOPPED);
+  protected inputTextValid$ = new BehaviorSubject<boolean>(false);
+  protected isValid$ = new BehaviorSubject<boolean>(false);
   protected readonly info: string[] = [];
 
+  private validators$: Array<Observable<boolean>> = [];
+  private validatorsSubscriptions$: Subscription = new Subscription();
+
+  private pipeline = new BehaviorSubject<estypes.IngestPipeline>({});
+
   constructor(
     protected readonly trainedModelsApi: ReturnType<typeof trainedModelsApiProvider>,
     protected readonly model: estypes.MlTrainedModelConfig,
     protected readonly inputType: INPUT_TYPE
   ) {
     this.modelInputField = model.input?.field_names[0] ?? DEFAULT_INPUT_FIELD;
     this.inputField = this.modelInputField;
+
+    this.validators$.push(
+      this.inputText$.pipe(map((inputText) => inputText.some((t) => t !== '')))
+    );
+  }
+
+  public destroy() {
+    this.validatorsSubscriptions$.unsubscribe();
+  }
+
+  protected initializeValidators(additionalValidators?: Array<Observable<boolean>>) {
+    if (additionalValidators) {
+      this.validators$.push(...additionalValidators);
+    }
+
+    this.validatorsSubscriptions$.add(
+      combineLatest(this.validators$)
+        .pipe(
+          map((validationResults) => {
+            return validationResults.every((v) => !!v);
+          })
+        )
+        .subscribe((v) => {
+          this.isValid$.next(v);
+          this.pipeline.next(this.generatePipeline());
```

Comment:
I think we should not mix validation logic with something else. Also here you generate a pipeline no matter if the input is valid or not. 

If I understand it correctly, the pipeline only depends on the model id and some additional params